ui/api/transport.go

```diff
@@ -2374,6 +2377,35 @@ func tokenFromCookie(r *http.Request, cookie string) (string, error) {
 	return c.Value, nil
 }
 
+func AdminAuthOMiddleware(next http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		tokenString, err := tokenFromCookie(r, "user")
+		if err != nil {
+			http.Redirect(w, r, "/"+errorsAPIEndpoint+url.QueryEscape(err.Error()), http.StatusSeeOther)
+			return
+		}
+
+		decodedUser, err := base64.StdEncoding.DecodeString(tokenString)
+		if err != nil {
+			http.Redirect(w, r, "/"+errorsAPIEndpoint+url.QueryEscape(err.Error()), http.StatusSeeOther)
+			return
+		}
+
+		var user sdk.User
+		if err = json.Unmarshal(decodedUser, &user); err != nil {
+			http.Redirect(w, r, "/"+errorsAPIEndpoint+url.QueryEscape(err.Error()), http.StatusSeeOther)
+			return
+		}
+
+		if user.Role != "admin" {
+			http.Redirect(w, r, "/"+errorsAPIEndpoint+url.QueryEscape(errors.ErrAuthorization.Error()), http.StatusSeeOther)
+			return
+		}
+
+		next.ServeHTTP(w, r)
+	})
```

Comment:
You can simplify with something like this (also remove the "/| prefix from of the errorsAPIEndpoint in this case of using fmt.Sprintf):
```suggestion
		var err error
		defer func() {
			if err != nil {
				http.Redirect(w, r, fmt.Sprintf("/%s/%s", errorsAPIEndpoint, url.QueryEscape(err.Error())), http.StatusSeeOther)
			}
		}()
		tokenString, err := tokenFromCookie(r, "user")
		if err != nil {
			return
		}

		decodedUser, err := base64.StdEncoding.DecodeString(tokenString)
		if err != nil {
			return
		}

		var user sdk.User
		if err = json.Unmarshal(decodedUser, &user); err != nil {
			return
		}

		if user.Role != "admin" {
			err = errors.ErrAuthorization
			return
		}

		next.ServeHTTP(w, r)
	})
```